.github/workflows/test-and-release.yaml

```diff
@@ -0,0 +1,116 @@
+name: Test and release
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - id: checkout
+        name: Checkout repository
+        uses: actions/checkout@v3
+
+      - id: setup-node
+        name: Setup Node.JS
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'yarn'
+
+      - id: install-deps
+        name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - id: lint
+        name: Lint code
+        run: yarn run eslint ./lib/ --ext .js --debug
+
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    needs: lint
+    strategy:
+      matrix:
+        node-version: ['14', '16', '18']
+    steps:
+      - id: checkout
+        name: Checkout repository
+        uses: actions/checkout@v3
+
+      - id: setup-node
+        name: Setup Node.JS
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'yarn'
+
+      - id: install-deps
+        name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - id: build
+        name: Build library
+        run: yarn build
+
+      - id: tests
+        name: Run tests
+        run: yarn test
+
+      - id: upload-artifact
+        name: Upload release candidate
+        uses: actions/upload-artifact@v3
+        with:
+          name: release-candidate
+          path: ./dist/
+
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    needs: test
+    permissions:
+      contents: write
+    # Restricts release to:
+    #   1) pushes of release tags
+    #   2) the default branch
+    #   3) the base repository
+    if: |
+      github.event_name == 'push' && startsWith(github.ref, 'refs/tags') &&
+      endsWith(github.event.base_ref, github.event.repository.default_branch) &&
+      github.repository == 'reuters-graphics/bluprint'
+    steps:
+      - id: checkout
+        name: Checkout repository
+        uses: actions/checkout@v3
+
+      - id: setup-node
+        name: Setup Node.JS
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          registry-url: https://registry.npmjs.org
+          scope: '@reuters-graphics'
+
+      - id: install-deps
+        name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - id: version
+        name: Version
+        run: npm version ${{ github.ref_name }} --no-git-tag-version
+
+      - id: publish
+        name: Publish
+        run: npm publish --access public
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - id: commit
+        name: Commit
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_message: Updated version to ${{ github.ref_name }}
+          branch: ${{ github.event.repository.default_branch }}
```

Comment:
Think we could get away without this dependency like I did [here](https://github.com/reuters-graphics/npm-ci-test/blob/main/.github/workflows/ci.yaml#L77-L83) if we add [a ref](https://github.com/reuters-graphics/npm-ci-test/blob/main/.github/workflows/ci.yaml#L52) to the checkout options. But maybe there's some other goodness this dude is giving us that I'm missing??

Reply:
If you've verified the git footwork can pull it off, sure, I'm good with it. I'll put that in.

Reply:
I've made that change. My only outstanding question is whether we can count on a naked `git push` to send changes to the default branch with this configuration.

Reply:
It did in testing, and I _think_ that's thanks to that [ref](https://github.com/reuters-graphics/npm-ci-test/blob/main/.github/workflows/ci.yaml#L52) option on checkout. But I suppose we could be double sure with:

```yaml
git push origin ${{ github.event.repository.default_branch }}
```

Reply:
If it works, it works. Cool config trick.